x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentToken.java

```diff
@@ -0,0 +1,215 @@
+                int httpCode = httpResponseApiKey.getHttpStatus();
+
+                if (httpCode != HttpURLConnection.HTTP_OK) {
+                    throw new UserException(ExitCodes.CANT_CREATE,
+                        "Unexpected response code [" + httpCode + "] from calling POST " + createAPIKeyURL(url));
+                }
+
+                final String apiKey = Objects.toString(httpResponseApiKey.getResponseBody().get("api_key"), "");
+                if (apiKey == null || apiKey.isEmpty()) {
+                    throw new UserException(ExitCodes.CANT_CREATE,
+                        "Could not create an api key.");
+                }
+
+                final HttpResponse httpResponseHttp = client.execute("GET", getHttpInfoURL(url), user, password,
+                    getHttpInfoRequestBodySupplier, is -> responseBuilder(is, terminal));
+                httpCode = httpResponseHttp.getHttpStatus();
+
+                if (httpCode != HttpURLConnection.HTTP_OK) {
+                    throw new UserException(ExitCodes.CANT_CREATE,
+                        "Unexpected response code [" + httpCode + "] from calling GET " + getHttpInfoURL(url));
+                }
+
+                final String address = getPublishedAddress(httpResponseHttp.getResponseBody());
+                if (address == null || address.isEmpty()) {
+                    throw new UserException(ExitCodes.CANT_CREATE,
+                        "Could not retrieve the ip address.");
+                }
+
+                final String stackVersion = getVersion(httpResponseHttp.getResponseBody());
+                if (stackVersion == null || stackVersion.isEmpty()) {
+                    throw new UserException(ExitCodes.CANT_CREATE,
+                        "Could not retrieve the version.");
+                }
+
+                final String fingerprint = SslUtil.calculateFingerprint(httpCaKeysAndCertificates.get(0).v2());
+
+                final XContentBuilder builder = JsonXContent.contentBuilder();
+                builder.startObject();
+                builder.field("version", stackVersion);
+                builder.startArray("adr");
+                builder.value(address);
+                builder.endArray();
+                builder.field("fgr", fingerprint);
+                builder.field("key", apiKey);//  CreateApiKeyResponse.getKey().toString());
+                builder.endObject();
+                final String jsonString = Strings.toString(builder);
+                return Base64.getUrlEncoder().encodeToString(jsonString.getBytes(StandardCharsets.UTF_8));
+            } catch (Exception e) {
+                logger.error(("Error generating enrollment token"), e);
+                throw new UserException(ExitCodes.CANT_CREATE, "Error generating enrollment token: " + e.getMessage());
+            }
+        }
+    }
+
+    private HttpResponse.HttpResponseBuilder responseBuilder(InputStream is, Terminal terminal) throws IOException {
```

Comment:
let's remove terminal.println() calls here. This class is not a CLI tool in itself, we should be logging instead